airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/util/S3OutputPathHelper.java

```diff
@@ -29,7 +29,13 @@ public static String getOutputPrefix(final String bucketPath, final String names
     if (namespace != null) {
       paths.add(NAME_TRANSFORMER.convertStreamName(namespace));
     }
-    paths.add(NAME_TRANSFORMER.convertStreamName(streamName));
+    if (streamName.contains("=")){
+      // case AWS Glue crawler
+      // more details https://docs.aws.amazon.com/glue/latest/dg/crawler-s3-folder-table-partition.html
+      paths.add(streamName);
+    }else {
+      paths.add(NAME_TRANSFORMER.convertStreamName(streamName));
```

Comment:
Shouldn't we create a `NamingConventionTransformer nameTransformer` when instantiating the `S3Destination`?

Here are some examples of how it's done in other connectors:
- Postgres: https://github.com/airbytehq/airbyte/blob/ea92a24da35ff9a32288bd96fc00a82ff7cad481/airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java#L38
- Snowflake: https://github.com/airbytehq/airbyte/blob/ea92a24da35ff9a32288bd96fc00a82ff7cad481/airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingDestination.java#L33
- MongoDB: https://github.com/airbytehq/airbyte/blob/ea92a24da35ff9a32288bd96fc00a82ff7cad481/airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongodbDestination.java#L64
- etc

Or at least, when we construct the consumer, we pass it to the factory:
- Redshift: https://github.com/airbytehq/airbyte/blob/ea92a24da35ff9a32288bd96fc00a82ff7cad481/airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftCopyS3Destination.java#L59

Reply:
I agree. We should hide that complexity in a custom NameTransformer

Reply:
As far as I understood, this converter is used for creating a stream name only.  A user wants to use some specific feature([AWS Glue crawler). That's why I removed any auto-replacement if "=" symbol is used

Reply:
@etsybaev the logic is sound, but we should encapsulate it into a `S3NameTransformer`